### PR TITLE
Feature/xaptc 40 GET a Test Suite

### PR DESCRIPTION
**Pull request by frankquinn**
Pulled from master and story 38 (PUT) so all relevant changes are included. No conflicts. All tests passing.

### CODE REVIEW

File: uiServices/src/testSuiteServices.go [hunk 1]
```diff
@@ -84,14 +87,100 @@ func ValidateJsonWithSchema(testSuite []byte, schemaName, structType string) boo
...
+
+	rw.WriteHeader(http.StatusNoContent)
+}
+
+func getTestSuite(rw http.ResponseWriter, req *http.Request){
+
+	testSuitePathDir := getTestSuiteHeader(req)
+	testSuiteName := chi.URLParam(req, "testSuiteName")
+
+	ValidateFileNameAndHeader(rw,req,testSuitePathDir, testSuiteName)
```
Comment (jenny-flynn) [inline uiServices/src/testSuiteServices.go]: check the response

File: uiServices/src/testSuiteServices.go [hunk 2]
```diff
@@ -84,14 +87,100 @@ func ValidateJsonWithSchema(testSuite []byte, schemaName, structType string) boo
...
+	if err != nil{
+		rw.WriteHeader(http.StatusInternalServerError)
+		logrus.Error("Cannot Read File")
+		return
+	}
+
+	err = xml.Unmarshal(byteValue, &testSuite)
+	if err != nil{
+		rw.WriteHeader(http.StatusInternalServerError)
+		logrus.Error("Cannot Unmarshall")
```
Comment (jenny-flynn) [inline uiServices/src/testSuiteServices.go]: change to "Cannot Unmarshall from XML" so we can pinpoint it

File: uiServices/src/testSuiteServices.go [hunk 3]
```diff
@@ -84,14 +87,100 @@ func ValidateJsonWithSchema(testSuite []byte, schemaName, structType string) boo
...
+		return
+	}
+
+	testSuiteJSON, err := json.MarshalIndent(testSuite,"","")
+	if err != nil {
+		rw.WriteHeader(http.StatusInternalServerError)
+		logrus.Error("Cannot Marshall")
```
Comment (jenny-flynn) [inline uiServices/src/testSuiteServices.go]: Cannot marshall to JSON

File: uiServices/src/testSuiteServices.go [hunk 4]
```diff
@@ -84,14 +87,100 @@ func ValidateJsonWithSchema(testSuite []byte, schemaName, structType string) boo
...
+		return
+	}
+
+	rw.WriteHeader(http.StatusOK)
+	rw.Write(testSuiteJSON)
+	logrus.Println(string(testSuiteJSON))
```
Comment (jenny-flynn) [inline uiServices/src/testSuiteServices.go]: remove println

File: uiServices/src/testSuiteServices_test.go [hunk 5]
```diff
@@ -191,3 +214,238 @@ func TestValidTestSuitePostConfigPathDirNotExist(t *testing.T) {
...
+	request.Header.Get("testSuitePathDir")
+
+	w := httptest.NewRecorder()
+	r.ServeHTTP(w, request)
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	assert.Equal(t, w.Code, http.StatusOK, "Error. Did not successfully GET")
```
Comment (jenny-flynn) [inline uiServices/src/testSuiteServices_test.go]: other way around. Keep your expected status to the left of your actual status code. If the test fails, the output will be misleading. Check other tests too (I've seen it in a few places)

File: uiServices/test/testSuite.feature [hunk 6]
```diff
@@ -152,4 +152,238 @@ Feature: Test Suite Creation
...
+    When I send "PUT" request to "/test-suites/GodogTestSuite" with body:
+  """
+      {
+       "testCases":[
+         {
+         "name":"file1.xml",
+          "preThinkTime": 2000,
+          "postThinkTime": 5000,
+          "execWeight": "Infrequent",
+           "description": "Desc1"
```
Comment (jenny-flynn) [inline uiServices/test/testSuite.feature]: is this necessary?

File: uiServices/test/testSuite.feature [hunk 7]
```diff
@@ -152,4 +152,238 @@ Feature: Test Suite Creation
...
+         }
+        ]
+      }
+    """
+    Then the response code should be 204
+    And the response body should be empty
+    When I send a "GET" request to "/test-suites/GodogTestSuite"
+    And the updated file should match json:
+"""
```
Comment (jenny-flynn) [inline uiServices/test/testSuite.feature]: line up the quotation marks

File: uiServices/src/testSuiteServices.go (hunk 1 above)
Comment (angerooney11) [reply uiServices/src/testSuiteServices.go]: What do you mean by this?

**Pull request merged**